File: Algo.Analytics/EmptyAnalyticsScript.cs
namespace StockSharp.Algo.Analytics
{
	/// <summary>
	/// The empty analytic strategy.
	/// </summary>
	public class EmptyAnalyticsScript : IAnalyticsScript
	{
		Task IAnalyticsScript.Run(ILogReceiver logs, IAnalyticsPanel panel, IEnumerable<Security> securities, DateTime from, DateTime to, IStorageRegistry storage, IMarketDataDrive drive, StorageFormats format, TimeSpan timeFrame, CancellationToken cancellationToken)
		{
			// !! add logic here

			return Task.CompletedTask;
		}
	}
}

File: Algo.Analytics.Python/empty_analytics_script.py
import clr

# Add .NET references
clr.AddReference("StockSharp.Messages")
clr.AddReference("StockSharp.Algo.Analytics")

from System import TimeSpan
from System.Threading.Tasks import Task
from StockSharp.Algo.Analytics import IAnalyticsScript
from storage_extensions import *

# The empty analytic strategy.
class empty_analytics_script(IAnalyticsScript):
    def Run(self, logs, panel, securities, from_date, to_date, storage, drive, format, time_frame, cancellation_token):
        if not securities:
            logs.LogWarning("No instruments.")
            return Task.CompletedTask

        # !! add logic here

        return Task.CompletedTask
